compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -24,6 +24,38 @@
 namespace
 {
 
+#define RET_FALSE_UNLESS(condition) \
+  if (not(condition))               \
+    return false;
+
+std::vector<int32_t> extract_shape(const luci::CircleNode *node)
+{
+  std::vector<int32_t> shape;
+  auto rank = node->rank();
+  for (auto i = 0u; i < rank; ++i)
+  {
+    shape.push_back(static_cast<int32_t>(node->dim(i).value()));
+  }
+  return shape;
+};
+
+template <loco::DataType DTYPE>
+std::vector<int32_t> extract_const(const luci::CircleConst *const_node)
+{
+  static_assert(DTYPE == loco::DataType::S32 || DTYPE == loco::DataType::S16 ||
+                  DTYPE == loco::DataType::S8,
```

Comment:
It seems unnecessary to allow S16 and S8.

```suggestion
  static_assert(DTYPE == loco::DataType::S32,
```